kong/pdk/service/response.lua

```diff
@@ -136,6 +139,28 @@ local function new(pdk, major_version)
   local MAX_HEADERS            = 1000
 
 
+  ---
+  -- Returns the upstream of the response from the Service as a string.
+  --
+  -- @function kong.service.response.get_upstream
+  -- @phases `header_filter`, `body_filter`, `log`
+  -- @treturn string|nil the ip:port from the response from the Service, or `nil`
+  -- @usage
+  -- kong.log.inspect(kong.service.response.get_upstream()) -- 127.0.0.1:80
+  function response.get_upstream()
+    check_phase(header_body_log)
+
+    if ngx.var.upstream_addr then
+      local addrs = split(ngx.var.upstream_addr, ',')
+      if #addrs > 0 then
+          return strip(addrs[#addrs])
```

Comment:
This needs a little bit more. According to http://nginx.org/en/docs/http/ngx_http_upstream_module.html ,

> If an internal redirect from one server group to another happens, initiated by `X-Accel-Redirect` or error_page, then the server addresses from different groups are separated by colons, e.g. `192.168.1.1:80, 192.168.1.2:80, unix:/tmp/sock : 192.168.10.1:80, 192.168.10.2:80`

My suggestion is also splitting by `:`:

```suggestion
      addrs = split(addrs, ':')
      if #addrs > 0 then
        return strip(addrs[#addrs])]
      end
```